File: sdks/python/ttsr.py
import requests

class TTSR:
    def __init__(self, base_url: str, client_id: str, client_secret: str):
        self.base = base_url.rstrip("/")
        self.client_id = client_id
        self.client_secret = client_secret
        self._token = None

    def _get_token(self, scope="score:write events:read"):
        if self._token:  # naive cache
            return self._token
        r = requests.post(f"{self.base}/v1/oauth2/token",
                          headers={"Content-Type":"application/x-www-form-urlencoded"},
                          data={"grant_type":"client_credentials","client_id":self.client_id,"client_secret":self.client_secret,"scope":scope})
        r.raise_for_status()
        self._token = r.json()["access_token"]
        return self._token

    def score(self, payload: dict, idempotency_key: str):
        tok = self._get_token("score:write")
        r = requests.post(f"{self.base}/v1/score",
                          headers={"Authorization":f"Bearer {tok}","Content-Type":"application/json","Idempotency-Key":idempotency_key},
                          json=payload)
        r.raise_for_status()
        return r.json(), r.headers

    def get_event(self, event_id: str):
        tok = self._get_token("events:read")
        r = requests.get(f"{self.base}/v1/events/{event_id}",
                         headers={"Authorization":f"Bearer {tok}"})
        r.raise_for_status()
        return r.json(), r.headers

File: sdks/js/ttsr.js
export class TTSR {
  constructor(baseUrl, clientId, clientSecret) {
    this.base = baseUrl.replace(/\/$/, "");
    this.clientId = clientId;
    this.clientSecret = clientSecret;
    this._token = null;
  }
  async _getToken(scope = "score:write events:read") {
    if (this._token) return this._token;
    const res = await fetch(`${this.base}/v1/oauth2/token`, {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: new URLSearchParams({
        grant_type: "client_credentials",
        client_id: this.clientId,
        client_secret: this.clientSecret,
        scope
      })
    });
    if (!res.ok) throw new Error(`Token failed ${res.status}`);
    const json = await res.json();
    this._token = json.access_token;
    return this._token;
  }
  async score(payload, idempotencyKey) {
    const tok = await this._getToken("score:write");
    const res = await fetch(`${this.base}/v1/score`, {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${tok}`,
        "Content-Type": "application/json",
        "Idempotency-Key": idempotencyKey
      },
      body: JSON.stringify(payload)
    });
    if (!res.ok) throw new Error(`Score failed ${res.status}`);
    return { json: await res.json(), headers: res.headers };
  }
  async getEvent(eventId) {
    const tok = await this._getToken("events:read");
    const res = await fetch(`${this.base}/v1/events/${eventId}`, {
      headers: { "Authorization": `Bearer ${tok}` }
    });
    if (!res.ok) throw new Error(`Get event failed ${res.status}`);
    return { json: await res.json(), headers: res.headers };
  }
}
